### PR TITLE
Adding a Java example using the V3 Margin Service API

### PR DESCRIPTION
**Pull request by andreiruse**
Adding an example class to showcase the newly added classes that are using the V3 API. 

This is supposed to be very similar to the previous MarginClientExample class, with the only notable difference being around getting CCP data.

### CODE REVIEW

File: modules/margin/src/main/java/com/opengamma/sdk/margin/v3/PortfolioDataFile.java [hunk 1]
```diff
@@ -95,6 +98,10 @@ public static PortfolioDataFile of(Path path) {
    * @throws UncheckedIOException if an IO error occurs
    */
   public static PortfolioDataFile ofCombined(List<Path> paths) {
+    if (paths.stream().anyMatch(Files::notExists)) {
```
Comment (cjkent) [inline modules/margin/src/main/java/com/opengamma/sdk/margin/v3/PortfolioDataFile.java]: You could build a nicer error message if you filtered on `Files::notExists` to get a collection of all non-existent files.

    List<Path> missingFiles = paths.stream().filter(Files::notExists).collect(toList());
    if (!missingFiles.isEmpty()) {
      String missingFileNames = missingFiles.stream().map(Object::toString).collect(joining(", "))
      throw new...

File: examples/src/main/java/com/opengamma/sdk/example/v3/MarginClientExample.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      // Creating the margin client
+      MarginClient client = MarginClient.of(invoker);
+
+      // Listing the CCPs that are available
+      CcpsResult ccps = client.listCcps();
+
+      // Optional step: Checking if we are permissioned to the specific CCP calculation engine
+      Ccp chosenCCP = Ccp.LCH;
+      if(!ccps.isCcpAvailable(chosenCCP)) {
+        throw new IllegalStateException(chosenCCP.name() + "Margin Calculator not available");
```
Comment (cjkent) [inline examples/src/main/java/com/opengamma/sdk/example/v3/MarginClientExample.java]: The error message will be a bit weird: "LCHMargin Calculator not available". Maybe:

    throw new IllegalStateException("Margin Calculator not available for " + chosenCCP.name());

File: examples/src/main/java/com/opengamma/sdk/example/v3/MarginClientExample.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+/**
+ * Example code for the OpenGamma Margin Service API (v3)
+ */
+public class MarginClientExample {
+
+  // credentials must be set to use the example
+  private static final Credentials CREDENTIALS = Credentials.ofApiKey("ABC", "123");
+
+  // the file to upload
+  private static final Path LCH_FILE = Paths.get("src/main/resources/lch-trades.txt");
```
Comment (cjkent) [inline examples/src/main/java/com/opengamma/sdk/example/v3/MarginClientExample.java:36]: Are users intended to run this or is it purely example code they will copy and paste and modify? 

Using a `Path` like this requires them to be in correct directory when they run it. Not sure if that's a problem.

**Review by cjkent — CHANGES_REQUESTED** [review]

File: examples/src/main/java/com/opengamma/sdk/example/v3/MarginClientExample.java (hunk 3 above)
Comment (andreiruse) [reply examples/src/main/java/com/opengamma/sdk/example/v3/MarginClientExample.java:36]: This is matching the previous example, for the V2 API.

This example is supposed to be runnable. Once the user has been provisioned with API credentials, this example will run as is. We also provide the sample file `lch-trades.txt` as part of this project.

Comment (cjkent) [reply examples/src/main/java/com/opengamma/sdk/example/v3/MarginClientExample.java:36]: In that case it's more reliable to use the resource mechanism to load the file. Something like:

    getClass().getResource("lch-trades.txt")

File: modules/margin/src/main/java/com/opengamma/sdk/margin/v3/PortfolioDataFile.java [hunk 4]
```diff
@@ -98,9 +99,10 @@ public static PortfolioDataFile of(Path path) {
    * @throws UncheckedIOException if an IO error occurs
    */
   public static PortfolioDataFile ofCombined(List<Path> paths) {
-    if (paths.stream().anyMatch(Files::notExists)) {
-      String firstFileNotFound = paths.stream().filter(Files::notExists).findFirst().get().toAbsolutePath().toString();
-      throw new IllegalArgumentException("Could not find one of the input files in the list." + firstFileNotFound);
+    List<Path> missingFiles = paths.stream().filter(Files::notExists).collect(Collectors.toList());
+    if (!missingFiles.isEmpty()) {
+      String missingFilesAsString = missingFiles.stream().map(Path::toString).collect(Collectors.joining(","));
```
Comment (cjkent) [inline modules/margin/src/main/java/com/opengamma/sdk/margin/v3/PortfolioDataFile.java]: Static import `Collectors` methods `toList` and `joining`.

**Review by cjkent — APPROVED** [review]

**Pull request merged**